tools/generator-go-sdk/generator/common_types.go

```diff
@@ -0,0 +1,142 @@
+package generator
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/hashicorp/pandora/tools/sdk/resourcemanager"
+	"github.com/hashicorp/pandora/tools/sdk/services"
+)
+
+type CommonTypesGenerator struct{}
+
+func NewCommonTypesGenerator() CommonTypesGenerator {
+	return CommonTypesGenerator{}
+}
+
+type SDKInput struct {
+	CommonPackageName         string
+	CommonPackageRelativePath string
+	CommonTypes               *services.ResourceManagerCommonTypes
+	OutputSubDirectoryName    string
+	OutputDirectoryPath       string
+	Source                    resourcemanager.ApiDefinitionsSource
+	VersionName               string
+}
+
+func (s *CommonTypesGenerator) GenerateForSDK(input SDKInput) error {
+	input.VersionName = strings.ToLower(input.VersionName)
```

Comment:
we should have a feature flag in here for `UseCommonTypes`? Presumably this can be fetched from the Data API (v2) since that has this toggle/information already - whilst that's not in v1 that would be pretty quick to add if needed